core/src/main/java/io/github/tcdl/msb/impl/RequesterImpl.java

```diff
@@ -109,6 +109,14 @@ public void publish(Object requestPayload, Message originalMessage, String tag)
         }
     }
 
+    private boolean isWaitForAckMs() {
+        return requestOptions.getAckTimeout() != null && requestOptions.getAckTimeout() != 0;
```

Comment:
It looks tricky. You work with fields of `RequestOptions` in a non-trivial way here and in `getWaitForResponsesFromConfigs`. The non-trivial semantics of values (if there's any) should be gathered in a single place.